process.go

```diff
@@ -280,13 +282,7 @@ func (p *process) createPeerController(ctx context.Context, peer route.Vertex,
 	return ctrl
 }
 
-func (p *process) eventLoop(ctx context.Context) error {
-	// Create a group to attach peer goroutines to.
-	group, ctx := errgroup.WithContext(ctx)
-	defer func() {
-		_ = group.Wait()
-	}()
-
+func (p *process) eventLoop(ctx context.Context, group *errgroup.Group) error {
```

Comment:
Originally I created a secondary group to keep `process` maximally self-contained. I think the change that you propose is effective.

I am wondering though if there is an easy way to avoid the group parameter here. I suppose that would be to also run the event loop itself as a member of the secondary group? This would get rid of the defer too I think and just make `group.Wait()` the last statement.

Curious to hear what you think of this option. If you feel it is over-engineered, it's fine to keep what you have.

Reply:
Tried to look up best practices for `errGroup`, couldn't find much. Docs say: "A Group is a collection of goroutines working on subtasks that are part of the same overall task." which makes me think that a second group is more canonical?

Also nice for the sake of having `eventLoop` completely terminate before returning (current impl lifting that wait up to the first group is a _slight_ behavior change ).